### PR TITLE
Valid move bishop

### PR DESCRIPTION
**Pull request by wookee55**
Lots of stuff going on in this commit because I started refactoring the code, then decided to create a separate branch for it, so I reverted the refactored code. For the most part, the only change is in bishop.rb, which is this:

    class Bishop < Piece
      def valid_moves
        BOARD.map do |file|
          next if file[0] == x_position && file[1] == y_position
          file if (file[0] - x_position).abs == (file[1] - y_position).abs
        end
      end
    end

### CODE REVIEW

**Comment by AUcod3r** [comment]
:+1:

File: app/controllers/games_controller.rb [hunk 1]
```diff
@@ -31,4 +31,4 @@ def update
   def game_params
     params.require(:game).permit(:users)
   end
-end
+end
```
Comment (hoitomt) [inline app/controllers/games_controller.rb]: Please update your text editor to add new lines at the end of the file. In SublimeText this is called `ensure_newline_at_eof_on_save` in Preferences -> Settings - Default

**Comment by hoitomt** [comment]
Looks good to me, nice work!

**Comment by wookee55** [comment]
Added the setting, seems like simply adds a new line to the end of the file, what is the advantage of this?

**Comment by wookee55** [comment]
Also, can someone please merge it into master?

**Comment by hoitomt** [comment]
@wookee55 Here is a good rationale for a new line at the end of the file - https://robots.thoughtbot.com/no-newline-at-end-of-file

**Pull request merged**